spec/ecraft/extensions/time_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'json'
+require 'ecraft/extensions/time'
+
+describe Date do
+  # More details: https://en.wikipedia.org/wiki/ISO_8601
+  describe '#to_json' do
+    context 'for values without a fraction part' do
+      it 'serializes the date and time in ISO 8601 form, not including a fraction part' do
+        json = { 'time' => Time.new(2000, 1, 1, 0, 0, 0, '+02:00') }.to_json
+        expect(json).to eq '{"time":"2000-01-01T00:00:00+02:00"}'
+      end
+    end
+
+    context 'for values with a fraction part' do
+      it 'serializes the date and time in ISO 8601 form, including the fraction part' do
+        json = { 'time' => Time.new(2000, 1, 1, 1, 2, 3.399, '+02:00') }.to_json
+        expect(json).to eq '{"time":"2000-01-01T01:02:03.399+02:00"}'
```

Comment:
Spec style: we could rely more on context, to have room for some concrete details ("milliseconds").

```ruby
describe '#to_json' do
  context 'serializes date and time to ISO8601 form' do
    subject(:json) { { 'time' => time }.to_json }

    context 'with a fraction part' do
      let(:time) { Time.new(2000, 1, 1, 1, 2, 3.399, '+02:00') }

      it 'renders the milliseconds' do
        expect(json).to eq '{"time":"2000-01-01T01:02:03.399+02:00"}'
      end
    end

    context 'without a fraction part' do
      let(:time) { Time.new(2000, 1, 1, 0, 0, 0, '+02:00') }

      it 'renders the milliseconds' do
        expect(json).to eq '{"time":"2000-01-01T00:00:00+02:00"}'
      end
    end
  end
end
```